test/gurka/test_landmarks.cc

```diff
@@ -288,6 +293,13 @@ TEST(LandmarkTest, TestParseLandmarks) {
 
   EXPECT_NO_THROW({ landmarks = db.get_landmarks_by_ids({3, 2, 1}); });
   EXPECT_EQ(landmarks.size(), 3);
+
+  // test round trip encoding
+  for (auto& expected : landmarks) {
+    expected.id = 0;
+    Landmark actual(expected.to_str());
+    EXPECT_TRUE(expected == actual);
+  }
```

Comment:
the landmark binary encoding had a bug in the lat lon transforms. to prove that i added this test which failed on amd64 as well. then i completely rewrote the encoding/decoding steps. the equality operator i made above uses approximate equality because there is a rounding that takes place on the 7th decimal place of the lat lons